packages/astro/src/content/runtime.ts

```diff
@@ -63,37 +67,55 @@ async function parseLiveEntry(
 	entry: LiveDataEntry,
 	schema: z.ZodType,
 	collection: string,
-): Promise<LiveDataEntry> {
-	const parsed = await schema.safeParseAsync(entry.data);
-	if (!parsed.success) {
-		throw new AstroError({
-			...AstroErrorData.InvalidContentEntryDataError,
-			message: AstroErrorData.InvalidContentEntryDataError.message(
-				collection,
-				entry.id,
-				parsed.error,
-			),
-		});
-	}
-	if (entry.cacheHint) {
-		const cacheHint = cacheHintSchema.safeParse(entry.cacheHint);
+): Promise<{ entry?: LiveDataEntry; error?: Error }> {
+	try {
+		const parsed = await schema.safeParseAsync(entry.data);
+		if (!parsed.success) {
+			return {
+				error: new AstroError({
```

Comment:
I'm not sure this should return an astro error. We usually keep these to be thrown, not to be used by users. What about a new `LiveLoaderError` type, like actions?

Reply:
Good idea